test/jest/network.spec.ts

```diff
@@ -0,0 +1,209 @@
+
+    const CURL_INTERNAL_8081 = [
+      "curl",
+      "-s",
+      "-o",
+      "/dev/null",
+      "-w",
+      "%{http_code}",
+      "http://curl-4.curl-test-2.svc.cluster.local:8081",
+    ];
+
+    // Deny request when port is not allowed on ingress
+    const denied_incorrect_port_response = await execInPod("test-admin-app", testAdminApp, "curl", CURL_INTERNAL_8081);
+    expect(denied_incorrect_port_response.stdout).toBe("503");
+
+    // Default Deny for undefined Ingress port
+    const blocked_port_curl = getCurlCommand("curl-4", "curl-test-2", 9999);
+    const denied_port_response = await execInPod("test-admin-app", testAdminApp, "curl", blocked_port_curl);
+    expect(denied_port_response.stdout).toBe("503");
+
+    // Wide open Egress means successful google curl
+    const successful_google_response = await execInPod("test-admin-app", testAdminApp, "curl", GOOGLE_CURL);
+    expect(successful_google_response.stdout).toBe("200");
+  });
+
+  test.concurrent("Anywhere Egress", async () => {
+    // Validate that request is successful when Egress Anywhere is used
+    const success_response = await execInPod("test-admin-app", testAdminApp, "curl", CURL_EXTERNAL);
+    expect(success_response.stdout).toBe("200");
+
+    // Validate Egress to Google is successful
+    const successful_google_response = await execInPod("test-admin-app", testAdminApp, "curl", GOOGLE_CURL);
+    expect(successful_google_response.stdout).toBe("200");
+  });
+
+  test.concurrent("RemoteNamespace Ingress and Egress", async () => {
+    // Validate that request is successful when using RemoteNamespace
+    const success_response = await execInPod("curl-test-4", curlPodName6, "curl-6", INTERNAL_CURL_COMMAND_5);
+    expect(success_response.stdout).toBe("200");
+
+    // Default Deny for Google Curl when no Egress defined
+    const denied_google_response = await execInPod("curl-test-4", curlPodName6, "curl-6", GOOGLE_CURL);
+    expect(denied_google_response.stdout).toBe("000");
+
+    // Default Deny for Blocked Port
+    const blocked_port_curl = getCurlCommand("curl-7", "curl-test-5", 9999);
+    const denied_port_response = await execInPod("curl-test-4", curlPodName6, "curl-6", blocked_port_curl);
+    expect(denied_port_response.stdout).toBe("503");
+  });
+
+  test.concurrent("Kube API Restrictions", async () => {
+    const kubeApi_curl = [
+      "curl",
+      "-s",
+      "-o",
+      "/dev/null",
+      "-w",
+      "%{http_code}",
+      "-k",
+      "-H",
+      "Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)",
+      `https://kubernetes.default.svc.cluster.local/api/v1/namespaces`,
+    ];
+
+    // Validate unauthorized response when using Kube API ( because of missing authorization token )
+    const success_unauthorized_response = await execInPod("curl-test-6", curlPodName8, "curl-8", kubeApi_curl);
+    expect(success_unauthorized_response.stdout).toContain("401");
```

Comment:
So there's a few issues here that cause the 401:
- The api endpoint we're trying to hit isn't allowed with our default svc account. You can however hit just the base API regardless of role: `https://kubernetes.default.svc.cluster.local/api`
- The `cat` command here is not getting evaluated in the right place (i.e. it's being evaluated on the host rather than in the pod). If you change the exec command to run inside a basic shell and wrap everything in single quotes this will work (execInPod cmd should end with `-- sh -c '${command.join(" ")}'``), but that might change if you want to try the client-node exec approach.

With those two changes locally I got this to respond with a 200 which feels better than checking for a 401 😄